### DISCUSSION TITLE
Should data from useQuery ever be undefined for cache-hit in react?

### DISCUSSION THREAD

**Discussion by austin43** [urql-graphql/urql#2329] [Q&A] [answered]
Hello,

I have a question about how graphcache operates with react.

Using debugger, I see the flow of a useQuery hook with a cacheHit go something like this:

initial render -> data: undefined, fetching: true
useQuery cache-first -> results in cache hit
second render -> data: cached result, fetching: false

This same flow repeats each time I render the view.

The expected behavior I have for a cache hit would be:
initial render -> useQuery cache-first -> results in cache hit -> data: cached result, fetching: false

Is the first scenario I described expected, or am I missing something?  My assumption would be that the cache is read synchronously, and the data always defined if there is a cache hit.  Thanks in advance for any comments, and for the continued work on this fantastic library :smile:

**Comment by JoviDeCroock** [comment, marked as answer]
Hmm this should only occur if you have an asynchronous exchange in front of your cacheExchange

**Reply by austin43** [reply]
That was it! I had authExchange in front of it.  Thank you for the quick reply.
